frame/evm/precompile/blake2/src/lib.rs

```diff
@@ -46,16 +49,21 @@ impl LinearCostPrecompile for Blake2F {
 
 		let mut rounds_buf: [u8; 4] = [0; 4];
 		rounds_buf.copy_from_slice(&input[0..4]);
-		let rounds: u32 = u32::from_le_bytes(rounds_buf);
+		let rounds: u32 = u32::from_be_bytes(rounds_buf);
+
+		let gas_cost: u64 = (rounds as u64) * Blake2F::GAS_COST_PER_ROUND;
+		if gas_cost > target_gas.expect("Gas limit not provided") { // TODO
+			return Err(ExitError::OutOfGas);
+		}
 
 		let mut h_buf: [u8; 64] = [0; 64];
-		h_buf.copy_from_slice(&input[4..48]);
+		h_buf.copy_from_slice(&input[4..68]);
 		let mut h = [0u64; 8];
 		let mut ctr = 0;
 		for state_word in &mut h {
 			let mut temp: [u8; 8] = Default::default();
-			temp.copy_from_slice(&h_buf[(ctr + 8)..(ctr + 1) * 8]);
-			*state_word = u64::from_le_bytes(temp).into();
+			temp.copy_from_slice(&h_buf[(ctr * 8)..(ctr + 1) * 8]);
+			*state_word = u64::from_ne_bytes(temp).into();
```

Comment:
Please use `from_le_bytes` instead. Yes, in both `amd64` and `wasm`, `ne` just means `le`, but we can't be sure that those are the only two building targets.

Reply:
Is this a requirement of Blake2? Would this work on big endian anyway?

Reply:
I think this is what is specified in the EIP spec.

Reply:
I follow. This has the effect of swapping endianness iff architecture is BE. Fixed in cb3e617 and feel free to critique my new comment.